core/src/main/java/io/paradoxical/cassieq/dataAccess/PointerRepositoryImpl.java

```diff
@@ -80,7 +95,7 @@ public PointerRepositoryImpl(Provider<Session> session, @Assisted QueueName queu
         return getOne(session.execute(query), mapper);
     }
 
-    private <T extends Pointer> T movePointer(PointerType pointerType, T original, T destination, Clause clause) {
+    private <T extends Pointer> boolean movePointer(PointerType pointerType, T destination, Clause clause) {
```

Comment:
consider this rename, as it will be clearer to the caller you should handle the case where it doesn't move.